sqlair.go

```diff
@@ -467,9 +472,14 @@ func (tx *TX) Query(ctx context.Context, s *Statement, inputArgs ...any) *Query
 		return &Query{ctx: ctx, err: ErrTXDone}
 	}
 
-	sqlstmt, err := stmtCache.prepareStmt(ctx, tx.db.cacheID, tx.sqlconn, s)
-	if err != nil {
-		return &Query{ctx: ctx, err: err}
+	sqlstmt, ok := stmtCache.lookupStmt(tx.db, s)
+	var err error
+	if !ok {
+		sqlstmt, err = tx.sqlconn.PrepareContext(ctx, s.te.SQL())
+		if err != nil {
+			return &Query{ctx: ctx, err: err}
+		}
+		stmtCache.storeStmt(tx.db, s, sqlstmt)
```

Comment:
I like this new abstraction a lot, it is now clear you are looking at the cache then falling back to preparing and then storing it in the cache.

Incidentally the problem with the connections is clearer now, you can see that we are preparing on `sqlconn` and then we are storing the statement based on `tx.db`.